### DISCUSSION TITLE
Control the speed of the voice an duration of pause between words/sentences

### DISCUSSION THREAD

**Discussion by kaosbeat** [erew123/alltalk_tts#181] [Q&A] [answered]
Hi,

Is there a way, preferably when making an API request, to lower the duration of pauses in between words and or sentences?

The voices are rather slow sometimes, triggering an "end of speech detected" when I STT the TTS text

**Comment by erew123** [comment, marked as answer]
Hi @kaosbeat 

The only real way to speed up pauses etc, that I know of with the XTTS model is to use a reference sample WAV that speaks faster. So if you have the same person speaking slower the generated text speaks slower (from anecdotal evidence) and faster when the reference sample wav is spoken a bit faster.

Beyond that, removing commas, semi-commas etc will remove pauses.

There is a generation speed, which speeds up the whole of the TTS generation, though Ive not played with it much myself.

https://docs.coqui.ai/en/latest/models/xtts.html#inference-parameters

You could manually introduce speed into `tts_server.py` by adding `"speed": 1.6,` (or a number of your choosing) by placing it in line 515

![image](https://github.com/erew123/alltalk_tts/assets/35898566/85b3d1a8-b722-4d3d-9b30-799a09236b0b)

```
        # Common arguments for both functions
        common_args = {
            "text": text,
            "language": language,
            "gpt_cond_latent": gpt_cond_latent,
            "speaker_embedding": speaker_embedding,
            "temperature": float(temperature),
            "length_penalty": float(model.config.length_penalty),
            "repetition_penalty": float(repetition_penalty),
            "speed": 1.6,
            "top_k": int(model.config.top_k),
            "top_p": float(model.config.top_p),
            "enable_text_splitting": True
        }
```

However, do note what it says in the Coqui documents about `(can produce artifacts if far from 1.0)`

Thanks

**Reply by kaosbeat** [reply]
Changing the "speed" to "1.3" Was adequate for me, thanks for pointing it out!
